### ISSUE TITLE
Two issues: (1) max_depth is not a recognized keyword argument and (2) depth returns a zero matrix

### ISSUE THREAD

**Issue by mavisvs** [DepthAnything/Depth-Anything-V2#265]
I am trying to run some images through the metric version of DepthAnythingV2, but for some reason max_depth is not a recognized keyword. Separately from that, I have an M1 chip, hence why I modified my model with .to('mps').

If I change both my model and map_location to 'mps' then the function runs but my depth returns a zero matrix...I'm not entirely sure what's wrong. Please let me know what other data would be helpful to provide. I'm pasting my code below for reference.

Before this issue, I was struggling to read the image files since my images are jpeg/jpg and don't have a shape (versus size). I changed this by converting my image into an np.array before inferring their depth. I also tried converting the image to a tensor using tf.convert_to_tensor as well. This is now commented out, though. This issue is now resolved, but I think it could be related to any of the aforementioned issues...

Thank you in advance~

------------------------
### importing packages
import cv2
import torch
import tensorflow as tf

from depth_anything_v2.dpt import DepthAnythingV2

#####################  
#importing the image from ISC backend
url = pd.read_csv('blank.csv')
flood_img = np.asarray(Image.open(requests.get(url, stream=True).raw))
#flood_img = tf.convert_to_tensor(flood_img)

#################

model_configs = {
    'vits': {'encoder': 'vits', 'features': 64, 'out_channels': [48, 96, 192, 384]},
    'vitb': {'encoder': 'vitb', 'features': 128, 'out_channels': [96, 192, 384, 768]},
    'vitl': {'encoder': 'vitl', 'features': 256, 'out_channels': [256, 512, 1024, 1024]}
}

encoder = 'vitl' # or 'vits', 'vitb'
dataset = 'vkitti' # 'hypersim' for indoor model, 'vkitti' for outdoor model
max_depth = 20 # 20 for indoor model, 80 for outdoor model

model = DepthAnythingV2(**{**model_configs[encoder], 'max_depth' : 80}).to('mps')#.to(torch.float32)
model.load_state_dict(torch.load(f'/Users/mstone/Depth-Anything-V2/Depth-Anything-V2/depth_anything_v2_metric_{dataset}_{encoder}.pth', map_location='mps'))
model.eval()

#raw_img = cv2.imread(image)
depth = model.infer_image(flood_img) # HxW depth map in meters in numpy

**Comment by heyoeyo** [comment]
For the first issue (max_depth is not a recognized), I think this is due to the model that's been imported. It's a bit confusing, but there's an inverse relative depth model and a metric model, and both use similar import paths. Only the metric model has the `max_depth` argument, so if you get that error, it means the relative depth model is being imported.
The simplest way to fix things would probably be to move your script into the [metric_depth folder](https://github.com/DepthAnything/Depth-Anything-V2/tree/main/metric_depth), since then the import pathing will target the metric depth version of the model. Otherwise, you could try changing the import pathing to something like:
```python
from metric_depth.depth_anything_v2.dpt import DepthAnythingV2
```

I'm not really sure about the second issue, but I'd guess that the result is ending up with `NaN` values (and/or infinities). So that might be something worth looking for (i.e. is the output all zeros, or are there other weird/invalid values in there?). In my experience, this usually happens when using float16 and only with certain images, switching to bfloat16 or float32 usually fixes things... I'm not familiar with the mac/mps side of things though and what options are available for dealing with this.
